src/functional/ffront/decorator.py

```diff
@@ -231,6 +239,44 @@ def _validate_args(self, *args, **kwargs) -> None:
             raise NotImplementedError("Keyword arguments are not supported yet.")
 
     def __call__(self, *args, offset_provider: dict[str, Dimension], **kwargs) -> None:
+        rewritten_args, size_args, kwargs = self._process_args(args, kwargs)
+
+        if not self.backend:
+            warnings.warn(
+                UserWarning(
+                    f"Field View Program '{self.itir.id}': Using default ({DEFAULT_BACKEND}) backend."
+                )
+            )
+        backend = self.backend if self.backend else DEFAULT_BACKEND
+
+        ensure_executor(backend)
+        if "debug" in kwargs:
+            debug(self.itir)
+
+        backend(
+            self.itir,
+            *rewritten_args,
+            *size_args,
+            **kwargs,
+            offset_provider=offset_provider,
+        )
+
+    def format_itir(
```

Comment:
I think this does not belong into the decorator, but in a dedicated utility function. It's sole tasks is (and believe should be) to create a callable object. The FieldOperator class is already too heavy right now.

Reply:
De facto the signature of the callable object it creates and owns is an implementation detail of `Program`, so it's responsibility is also to be a facade to that callable.

If we want to free it from this facade responsibility, then it's callable object has to own the `_process_args` functionality.

Reply:
... After due consideration, I think it is not a refactoring that fits into this scope. So I added a TODO.